src/dotty/tools/dotc/core/pickling/PickleFormat.scala

```diff
@@ -0,0 +1,454 @@
+                  SHORTneg             NegNat
+                  CHARconst            Nat
+                  INTconst             Nat
+                  INTneg               NegNat
+                  LONGconst            LongNat
+                  LONGneg              NegLongNat
+                  FLOATconst           LongNat
+                  DOUBLEconst          LongNat
+                  STRINGconst          NameRef
+                  NULLconst
+                  CLASSconst    Length Type
+                  ENUMconst     Length Path
+  NegNat        = Nat          					// negValue = -natValue - 1
+  NegLongNat    = LongNat     					// negValue = -natValue - 1
+
+  Type          = Path
+                  TYPEREFdirect        sym_ASTRef
+                  TYPEREFstatic        fullyQualified_NameRef
+                  TYPEREFsymbol        qual_Type sym_ASTRef
+                  TYPEREF              qual_Type possiblySigned_NameRef
+                  SUPERtype     Length this_Type underlying_Type
+                  SKOLEMtype    Length underlying_Type
+                  REFINEDtype   Length refinement_NameRef info_Type
+                  APPLIEDtype   Length tycon_Type arg_Type*
+                  TYPEBOUNDS    Length low_Type high_Type
+                  TYPEALIAS     Length alias_Type
+                  ANNOTATED     Length annot_Tree underlying_Type
+                  ANDtype       Length left_Type right_Type
+                  ORtype        Length left_Type right_Type
+                  BIND          Length boundName_NameRef underlying_Type selfRef_Type
+                  BYNAMEtype    Length underlying_Type
+                  POLYtype      Length result_Type NamesTypes      // needed for refinements
+                  METHODtype    Length result_Type NamesTypes      // needed for refinements
+                  PARAMtype     Length binder_ASTref paramNum_Nat  // needed for refinements
+                  NOTYPE
+                  NOPREFIX
+                  SHARED               type_ASTRef
+  NamesTypes    = ParamType*
+  NameType      = paramName_NameRef typeOrBounds_ASTRef
+
+  Modifier      = PRIVATE
+                  INTERNAL						// package private
+                  PROTECTED
+                  PRIVATEqualified     Length qualifier_Type		// will be dropped
+                  PROTECTEDqualified   Length qualifier_Type 	// will be dropped
+                  ABSTRACT
+                  FINAL
+                  SEALED
+                  CASE
+                  IMPLICIT
+                  LAZY
+                  OVERRIDE
+                  INLINE
+                  ABSOVERRIDE					// abstract override
+                  STATIC						// mapped to static Java member
+                  MODULE						// an object or its class
+                  LOCAL						// private[this] or protected[this]
+                  SYNTHETIC					// generated by Scala compiler
+                  ARTIFACT						// to be tagged Java Synthetic
```

Comment:
What's the difference between SYNTHETIC and ARTIFACT? Always been wondering about this wrt scalac.